test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -885,8 +888,25 @@ public static String randomGeohash(int minPrecision, int maxPrecision) {
         return geohashGenerator.ofStringLength(random(), minPrecision, maxPrecision);
     }
 
-    public static String randomTestTransport() {
-        return randomBoolean() ? NioTransportPlugin.NIO_TRANSPORT_NAME : MockTcpTransportPlugin.MOCK_TCP_TRANSPORT_NAME;
+    private static final SetOnce<Boolean> useNio = new SetOnce<>();
```

Comment:
Do you mean:

```
private static final boolean useNio = randomBoolean();
```

Not sure how the other would even compile...but it should work, `randomBoolean()` is a static method.